File: Excepciones/ejercicio 6/CuentaBancaria.java
public class CuentaBancaria {
    private String numeroCuenta;
    private String titular;
    private double saldo;

    public CuentaBancaria(String numeroCuenta, String titular, double saldoInicial) {
        this.numeroCuenta = numeroCuenta;
        this.titular = titular;
        this.saldo = saldoInicial;
    }

    public void depositar(double monto) {
        if (monto <= 0) {
            throw new IllegalArgumentException("El monto a depositar debe ser positivo.");
        }
        saldo += monto;
        System.out.println("Depósito realizado: " + monto);
    }
    public void retirar(double monto) throws FondosInsuficientesException {
        if (monto > saldo) {
            throw new FondosInsuficientesException(
                    "Fondos insuficientes. Saldo disponible: " + saldo + ", intento de retiro: " + monto);
        }
        saldo -= monto;
        System.out.println("Retiro realizado: " + monto);
    }
    public void mostrarInfo() {
        System.out.println("Número de cuenta: " + numeroCuenta);
        System.out.println("Titular: " + titular);
        System.out.println("Saldo: " + saldo);
        System.out.println();
    }
}

File: Excepciones/ejercicio 6/CuentaBancaria.py
class FondosInsuficientesException(Exception):
    pass
class CuentaBancaria:
    def __init__(self, numero, titular, saldo):
        self.numero = numero
        self.titular = titular
        self.saldo = saldo

    def depositar(self, monto):
        if monto <= 0:
            raise ValueError("El monto debe ser positivo.")
        self.saldo += monto

    def retirar(self, monto):
        if monto > self.saldo:
            raise FondosInsuficientesException("Fondos insuficientes.")
        self.saldo -= monto

    def mostrar_info(self):
        print(f"Número: {self.numero}, Titular: {self.titular}, Saldo: {self.saldo}")

cuenta = CuentaBancaria("12345", "Juan Pérez", 1000)

try:
    cuenta.depositar(500)
    cuenta.depositar(-20)
except ValueError as e:
    print("Error en depósito:", e)

try:
    cuenta.retirar(2000)
except FondosInsuficientesException as e:
    print("Error en retiro:", e)
